control_server/httpserver/routes.go

```diff
@@ -0,0 +1,142 @@
+package httpserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"syscall"
+
+	"github.com/gorilla/mux"
+
+	"baas/pkg/fs"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/google/uuid"
+
+	"baas/control_server/machines"
+	"baas/pkg/api"
+	"baas/pkg/model"
+)
+
+// Routes is a struct on which functions are defined that respond to requests
+// from the management OS. This struct holds state necessary for the request handlers.
+type Routes struct {
+	machineStore machines.MachineStore
+	diskpath     string
+}
+
+// BootInform handles all incoming boot inform requests
+func (routes *Routes) BootInform(w http.ResponseWriter, r *http.Request) {
+	var bootInform api.BootInformRequest
+
+	if err := json.NewDecoder(r.Body).Decode(&bootInform); err != nil {
+		log.Errorf("Error while parsing json: %v", err)
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	log.Debug("Received BootInform request, serving Reprovisioning information")
+
+	// handle things based on bootinform
+
+	// Request data from database for what to do with this machine
+	uuid1 := "alpineresult.iso"
+	uuid2 := "alpine.iso"
+	location := "/dev/sda"
+
+	// Prepare response
+	resp := api.ReprovisioningInfo{
+		Prev: model.MachineSetup{
+			Ephemeral: false,
+			Disks: map[model.DiskUUID]model.DiskImage{
+				uuid1: {
+					DiskType:             model.DiskTypeRaw,
+					DiskTransferStrategy: model.DiskTransferStrategyHTTP,
+					Location:             location,
+				},
+			},
+		},
+		Next: model.MachineSetup{
+			Ephemeral: false,
+			Disks: map[model.DiskUUID]model.DiskImage{
+				uuid2: {
+					DiskType:             model.DiskTypeRaw,
+					DiskTransferStrategy: model.DiskTransferStrategyHTTP,
+					Location:             location,
+				},
+			},
+		},
+	}
+
+	if err := json.NewEncoder(w).Encode(&resp); err != nil {
+		log.Errorf("Error while serialising json: %v", err)
+		http.Error(w, "Error while serialising response json", http.StatusInternalServerError)
+		return
+	}
+
+	r.Header.Set("content-type", "application/json")
+}
+
+// UploadDiskImage allows the management os to upload disk images
+func (routes *Routes) UploadDiskImage(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id, ok := vars["uuid"]
+	if !ok || id == "" {
+		http.Error(w, "Invalid uuid", http.StatusBadRequest)
+		log.Error("Invalid uuid given")
+		return
+	}
+
+	path := fmt.Sprintf("%s/%s", routes.diskpath, id)
+	temppath := path + "." + uuid.New().String() + ".tmp"
```

Comment:
use `Sprintf` for consistency? (not very important)

Reply:
fixed